google/cloud/storage/storage_client_options_test.cc

```diff
@@ -0,0 +1,99 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/internal/setenv.h"
+#include "google/cloud/storage/client_options.h"
+#include "google/cloud/testing_util/environment_variable_restore.h"
+#include <gmock/gmock.h>
+
+namespace google {
+namespace cloud {
+namespace storage {
+inline namespace STORAGE_CLIENT_NS {
+namespace {
+class ClientOptionsTest : public ::testing::Test {
+ public:
+  ClientOptionsTest()
+      : enable_tracing_("CLOUD_STORAGE_ENABLE_TRACING"),
+        endpoint_("CLOUD_STORAGE_TESTBENCH_ENDPOINT") {}
+
+ protected:
+  void SetUp() override { enable_tracing_.SetUp(); }
+  void TearDown() override { enable_tracing_.TearDown(); }
+
+ protected:
+  testing_util::EnvironmentVariableRestore enable_tracing_;
+  testing_util::EnvironmentVariableRestore endpoint_;
+};
+
+TEST_F(ClientOptionsTest, Default) {
+  google::cloud::internal::SetEnv("CLOUD_STORAGE_ENABLE_TRACING", nullptr);
+  // Create the options with the insecure credentials because the default
+  // credentials may result in an error in the CI environment.
```

Comment:
Why would the default credentials result in such an error?

Reply:
Because the default credentials try to load a credentials file, and those may not (and should not) exist in the CI environment.